src/golang/lib/databricks/jobs.go

```diff
@@ -45,71 +45,122 @@ func CreateJob(
 	ctx context.Context,
 	databricksClient *databricks_sdk.WorkspaceClient,
 	name string,
+	tasks []jobs.JobTaskSettings,
+) (int64, error) {
+
+	createRequest := &jobs.CreateJob{
+		Name:  name,
+		Tasks: tasks,
+	}
+	createResp, err := databricksClient.Jobs.Create(ctx, *createRequest)
+	if err != nil {
+		return -1, errors.Wrap(err, "Error creating a job in Databricks.")
+	}
+	return createResp.JobId, nil
+}
+
+func CreateTask(
+	ctx context.Context,
+	databricksClient *databricks_sdk.WorkspaceClient,
+	name string,
+	upstreamTaskNames []string,
 	s3InstanceProfileArn string,
 	pythonFilePath string,
-) (int64, error) {
+	specStr string,
+) (*jobs.JobTaskSettings, error) {
+
 	sparkVersions, err := databricksClient.Clusters.SparkVersions(ctx)
 	if err != nil {
-		return -1, errors.Wrap(err, "Error selecting a spark version.")
+		return nil, errors.Wrap(err, "Error selecting a spark version.")
 	}
-
 	// Select the latest LTS version.
 	latestLTS, err := sparkVersions.Select(clusters.SparkVersionRequest{
 		Latest:          true,
 		LongTermSupport: true,
 	})
 	if err != nil {
-		return -1, errors.Wrap(err, "Error selecting a spark version.")
+		return nil, errors.Wrap(err, "Error selecting a spark version.")
+	}
+	taskDependenciesList := make([]jobs.TaskDependenciesItem, 0, len(upstreamTaskNames))
+	for _, taskName := range upstreamTaskNames {
+		taskDependenciesList = append(taskDependenciesList, jobs.TaskDependenciesItem{TaskKey: taskName})
 	}
 
-	createRequest := &jobs.CreateJob{
-		Name: name,
-		Tasks: []jobs.JobTaskSettings{
+	task := &jobs.JobTaskSettings{
+		TaskKey: name,
+		NewCluster: &clusters.CreateCluster{
+			SparkVersion: latestLTS,
+			NumWorkers:   NumWorkers,
+			NodeTypeId:   NodeTypeId,
+			AwsAttributes: &clusters.AwsAttributes{
+				InstanceProfileArn: s3InstanceProfileArn,
+			},
+		},
+		DependsOn: taskDependenciesList,
+		SparkPythonTask: &jobs.SparkPythonTask{
+			PythonFile: pythonFilePath,
+			Parameters: []string{"--spec", specStr},
+		},
+		Libraries: []libraries.Library{
 			{
-				TaskKey: name,
-				NewCluster: &clusters.CreateCluster{
-					SparkVersion: latestLTS,
-					NumWorkers:   NumWorkers,
-					NodeTypeId:   NodeTypeId,
-					AwsAttributes: &clusters.AwsAttributes{
-						InstanceProfileArn: s3InstanceProfileArn,
-					},
+				Pypi: &libraries.PythonPyPiLibrary{
+					Package: fmt.Sprintf("aqueduct-ml==%s", lib.ServerVersionNumber),
 				},
-				SparkPythonTask: &jobs.SparkPythonTask{
-					PythonFile: pythonFilePath,
-				},
-				Libraries: []libraries.Library{
-					{
-						Pypi: &libraries.PythonPyPiLibrary{
-							Package: fmt.Sprintf("aqueduct-ml==%s", lib.ServerVersionNumber),
-						},
-					},
+			},
+			{
+				Pypi: &libraries.PythonPyPiLibrary{
+					Package: "snowflake-sqlalchemy",
```

Comment:
Why is the `snowflake` package hardcoded?

Reply:
This needs to be updated. Thanks for the catch!

Reply:
Is this going to be updated as part of this PR?

Reply:
Going to do this in a follow up PR.